File: backend/src/models/implementations/echoModel.js
const BaseModel = require('../baseModel');

/**
 * EchoModel - A simple model that echoes back the user's message
 */
class EchoModel extends BaseModel {

  constructor() {
    super(BaseModel.types.CHAT, 'Echo');
  }

  /**
   * Generates a response based on the user message
   * @param {Object} input - Input containing userMessage and conversationHistory
   * @returns {Object} Response object with actor and content
   */
  async process(input) {
    const { userMessage, conversationHistory } = input;
    return {
      actor: 'model',
      content: `You said: "${userMessage}" - Hello from the Echo Model!`
    };
  }
}

module.exports = EchoModel;

File: backend-py/src/models/implementations/echoModel.py
from src.models.baseModel import BaseModel, ModelType

class EchoModel(BaseModel):
    """
    EchoModel - A simple model that echoes back the user's message
    """
    
    def __init__(self):
        """
        Initialize a new EchoModel instance
        """
        super().__init__(ModelType.CHAT, "echo")
    
    async def process(self, input_data):
        """
        Generates a response based on the user message
        
        Args:
            input_data (dict): Input containing userMessage and conversationHistory
            
        Returns:
            dict: Response object with actor and content
        """
        user_message = input_data.get("userMessage", "")
        return {
            "actor": "model",
            "content": f'You said: "{user_message}" - Hello from the Python Echo Model!'
        }
